### PR TITLE
perf: String#getBytes(Charset) vs getBytes(String)

### PR DESCRIPTION
**Pull request by tamird**
rebase of #18. Closes #18.

### CODE REVIEW

**Comment by tamird** [comment]
I believe @phraktle signed the CLA back in #18.

**Comment by phraktle** [comment]
Yes, I'm happy to contribute this code under the CLA.

**Comment by tamird** [comment]
@xfxyjwf green!

**Comment by tamird** [comment]
@pherl could you take a look please?

File: java/src/main/java/com/google/protobuf/Internal.java [hunk 1]
```diff
@@ -51,6 +52,9 @@
  * @author kenton@google.com (Kenton Varda)
  */
 public class Internal {
+
+  protected static final Charset ISO_8859_1 = Charset.forName("ISO-8859-1");
```
Comment (xfxyjwf) [inline java/src/main/java/com/google/protobuf/Internal.java]: Could you move the two constants UTF_8 and ISO_8859_1 in the same class?

File: java/src/test/java/com/google/protobuf/LiteralByteStringTest.java [hunk 2]
```diff
@@ -293,14 +291,21 @@ public void testNewOutput() throws IOException {
 
   public void testToString() throws UnsupportedEncodingException {
     String testString = "I love unicode \u1234\u5678 characters";
-    LiteralByteString unicode = new LiteralByteString(testString.getBytes(UTF_8));
-    String roundTripString = unicode.toString(UTF_8);
+    LiteralByteString unicode = new LiteralByteString(testString.getBytes(ByteString.UTF_8));
```
Comment (xfxyjwf) [inline java/src/test/java/com/google/protobuf/LiteralByteStringTest.java]: This test seems exactly same as the testCharsetToString(). I suppose you meant to test unicode.toString("UTF-8") in this test case? The test for RopeByteString has the same problem.

Comment (tamird) [inline java/src/test/java/com/google/protobuf/LiteralByteStringTest.java]: Good catch! Fixed.

File: java/src/main/java/com/google/protobuf/Internal.java (hunk 1 above)
Comment (tamird) [inline java/src/main/java/com/google/protobuf/Internal.java]: Done - both are in `Internal` now

**Comment by tamird** [comment]
Green

**Comment by xfxyjwf** [comment]
Thanks!

**Pull request merged**